extensions/mssql/src/features.ts

```diff
@@ -43,7 +43,12 @@ export class AccountFeature implements StaticFeature {
 		let timeToLiveInSeconds = 10;
 		this.tokenCache = new DataItemCache(this.getToken, timeToLiveInSeconds);
 		this._client.onRequest(contracts.SecurityTokenRequest.type, async (request): Promise<contracts.RequestSecurityTokenResponse | undefined> => {
-			return this.tokenCache.getData(request);
+			return await this.tokenCache.getData(request);
+		});
+		this._client.onNotification(contracts.RefreshTokenNotification.type, async (request) => {
+			// send refresh token to sts to get updated
+			let result = await this.refreshToken(request);
```

Comment:
Have you verified what happens if an error occurs during the refresh process? Is the user going to know when something unexpected happened and recover from it?